Apps/Common/src/AspNetConfiguration/Modules/OdrConfiguration.cs

```diff
@@ -0,0 +1,80 @@
+// -------------------------------------------------------------------------
+//  Copyright © 2019 Province of British Columbia
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+// -------------------------------------------------------------------------
+namespace HealthGateway.Common.AspNetConfiguration.Modules
+{
+    using System;
+    using System.Diagnostics.CodeAnalysis;
+    using System.Text.Json;
+    using System.Text.Json.Serialization;
+    using HealthGateway.Common.Data.Utils;
+    using HealthGateway.Common.Models.ODR;
+    using HealthGateway.Common.Utils.Odr;
+    using Microsoft.Extensions.Configuration;
+    using Microsoft.Extensions.DependencyInjection;
+    using Refit;
+
+    /// <summary>
+    /// Provides ASP.Net Services for ODR access.
+    /// </summary>
+    [ExcludeFromCodeCoverage]
+    public static class OdrConfiguration
+    {
+        /// <summary>
+        /// Configures the Refit API for accessing ODR services and adds it to the services collection.
+        /// </summary>
+        /// <param name="services">The service collection provider.</param>
+        /// <param name="configuration">The configuration to use for values.</param>
+        /// <typeparam name="TApi">The Refit API to be configured.</typeparam>
+        public static void AddOdrRefitClient<TApi>(IServiceCollection services, IConfiguration configuration)
+            where TApi : class
```

Comment:
why is this needed and why is the method generic? HG services should be able to communicate to ODR without changing anything but the url of the remote service and a little configuration?

I'd expect a different options class to modify the httpclient handler that refit will use when connecting to the remote endpoint

Reply:
This was a refactor to avoid duplicating code in Encounter and Medication.  It's generic because we split the ODR calls into separate Refit APIs.  We could easily merge the Refit APIs and move them to Common instead.

Reply:
I think it is better. Even better, move ODR to its own assembly including the adapter/delegate so the usable unit is the assembly, not code.